workers/unity/Assets/Playground/Scripts/EntityInitialization/ArchetypeInitializationSystem.cs

```diff
@@ -67,47 +55,9 @@ protected override void OnUpdate()
             {
                 var archetypeName = data.ArchetypeComponents[i].ArchetypeName;
                 var entity = data.Entities[i];
-
-                if (!ArchetypeConfig.WorkerTypeToArchetypeNameToComponentTypes[worker.WorkerType]
-                    .TryGetValue(archetypeName, out var componentTypesToAdd))
-                {
-                    worker.LogDispatcher.HandleLog(LogType.Error, new LogEvent(ArchetypeMappingNotFound)
-                        .WithField(LoggingUtils.LoggerName, LoggerName)
-                        .WithField("ArchetypeName", archetypeName)
-                        .WithField("WorkerType", worker.WorkerType));
-                    continue;
-                }
-
-                foreach (var componentType in componentTypesToAdd)
-                {
-                    var type = componentType.GetManagedType();
-                    var componentInstance = Activator.CreateInstance(type);
-
-                    if (!typeToIsComponentData.TryGetValue(type, out var isComponentData))
-                    {
-                        isComponentData = type.GetInterfaces().Contains(typeof(IComponentData));
-                        typeToIsComponentData[type] = isComponentData;
-                    }
-
-                    if (isComponentData)
-                    {
-                        if (!typeToAddComponentGenericMethodInfo.TryGetValue(type, out var addComponentMethodGeneric))
-                        {
-                            addComponentMethodGeneric = addComponentMethod.MakeGenericMethod(type);
-                            typeToAddComponentGenericMethodInfo[type] = addComponentMethodGeneric;
-                        }
-
-                        addComponentMethodGeneric.Invoke(PostUpdateCommands,
-                            new[] { entity, componentInstance });
-                    }
-                    else
-                    {
-                        viewCommandBuffer.AddComponent(entity, componentType, componentInstance);
-                    }
-                }
-            }
-
-            viewCommandBuffer.FlushBuffer();
+                
+                ArchetypeConfig.AddComponentDataForArchetype(worker, archetypeName, PostUpdateCommands, entity);
```

Comment:
It feels strange that we call methods on a "config" directly.

Reply:
@jamiebrynes7 Don't we do the same with `SystemConfig` as well?

Reply:
I think we do, but we shouldn't to be fair. Need to think for better names for these

Reply:
actually we only use that method in one place, so might just move it directly into the system